src/System.Text.Primitives/System/Text/EncodingData.cs

```diff
@@ -365,8 +345,153 @@ private bool VerifySuffix(ReadOnlySpan<byte> buffer, int codeUnitIndex, Symbol s
         // CC = the max value, and DD = the max value's index in the same coord-system as BB.
         struct ParsingTrieNode
         {
-            public byte valueOrNumChildren;
+            public byte ValueOrNumChildren;
             public int IndexOrSymbol;
         }
+        
```

Comment:
For the separate file, would you also want a separate type? Or is it fine to keep this in EncodingData?